diff --git a/dlls/cryptnet/cryptnet_main.c b/dlls/cryptnet/cryptnet_main.c
index 315432aa108bbce820e41cc4c6a46e07c28851e7..4432f38e5b84f3a8b3772d5971354ad1f24b39d5 100644
--- a/dlls/cryptnet/cryptnet_main.c
+++ b/dlls/cryptnet/cryptnet_main.c
@@ -1641,6 +1641,7 @@ static BOOL find_cached_revocation_status(const CERT_CONTEXT *cert, const CERT_R
     }
 
     TRACE("Using cached status %#lx, reason %#lx.\n", status->dwError, status->dwReason);
+    fclose(file);
     return TRUE;
 }
 
